File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.go
func reverseWords(s string) string {
    var ans, temp string

    for _, char := range s {
        if char == ' ' {
            ans += reverseString(temp)
            ans += " "
            temp = ""
        } else {
            temp += string(char)
        }
    }

    ans += reverseString(temp)

    return ans
}

func reverseString(s string) string {
    runes := []rune(s)
    length := len(runes)

    for i := 0; i < length/2; i++ {
        runes[i], runes[length-1-i] = runes[length-1-i], runes[i]
    }

    return string(runes)
}

File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.rs
impl Solution {
    pub fn reverse_words(s: String) -> String {
        let mut ans = String::new();
        let mut temp = String::new();

        for i in s.chars() {
            if i == ' ' {
                ans += &temp.chars().rev().collect::<String>();
                ans += " ";
                temp.clear();
            } else {
                temp.push(i);
            }
        }

        ans += &temp.chars().rev().collect::<String>();

        ans
    }
}